src/yagolic.c

```diff
@@ -11,26 +11,26 @@ int main() {
   scanf("%d", &rows); // NOLINT acanf_s is not implemented in gnu c standard
   printf("How many COLUMNS shall your board have ?\n");
   scanf("%d", &cols); // NOLINT acanf_s is not implemented in gnu c standard
-  while ((sc = getchar()) != '\n' && sc != EOF)
+  while ((user_input = getchar()) != '\n' && user_input != EOF)

```

Comment:
Why do we need this while loop here?

Reply:
We don't actually, and removing it removed the need for the variable, see the new commit for the related changes.